core/trino-spi/src/main/java/io/trino/spi/predicate/NullableValue.java

```diff
@@ -43,7 +45,9 @@ public NullableValue(Type type, Object value)
     {
         requireNonNull(type, "type is null");
         if (value != null && !Primitives.wrap(type.getJavaType()).isInstance(value)) {
-            throw new IllegalArgumentException(format("Object '%s' does not match type %s", value, type.getJavaType()));
+            if (!(type instanceof DateType && value instanceof LongTimestampWithTimeZone)) {
+                throw new IllegalArgumentException(format("Object '%s' does not match type %s", value, type.getJavaType()));
```

Comment:
We shouldn't need to have this SPI change to make Iceberg connector work.
The code here was correct.

Reply:
But after casting it goes to the Timestamp branch, what shall I do for this? @findepi